Chase_step1b_AssignOutcome_Algorithm3.sas

```diff
@@ -70,104 +97,143 @@ data &discdsn._int_Alg3 ;
+	**Check for code type (proc/diag/rx) and assign outcome based on Hierarchy for algorithm hier(*);
 
+	*Assign outcomes;
+	*Concordant pregnancy outcome groups receive the concordant outcome;
+ 	if concordant then outassgn_pt1= outcome_concordant; *3b.2;
 
-proc freq data= &newdsn._alg3;*where outcome_assigned3='unk';
-where outassgn_pt1='UAB' and outcome_assigned3 ne 'UAB';
-*table LBM* LBS* MLS* SB* UDL* SAB* IAB* UAB* EM* AEM /list missing;
-*table delivery* SAB* IAB* UAB* EM* AEM *pr_n*outassgn_pt1* discordant3 /list missing;
-table delivery* SAB* IAB* UAB* EM* AEM  * discordant3 *outcome_assigned3 /list missing;*uab-pr other ab not pr;
-format pr_n onepl.;
-run;
+ 	else do;
+
+  		*step 1b.3 discordant but with concordant delivery procedure using table 1 (Disc_OutPot3= PRD-[x]);
+  		if discordant3=1 then outassgn_pt1= substr(disc_outpot3,5); 
+
+  		*step 1b.4 discordant with delivery discordant;
+  		if discordant3=2 then outassgn_pt1= 'UDL'; *discordant=udl per instruction substr(disc_outpot3,5);
 
+  		*step 1b.5 discordant, no delivery;
+  		if discordant3=3 then do ;
+
+     		*using hierarchy for 1b.5 - first prioritize procedure codes;
+     		if pr_n>=1 then do i=1 to 10 until (outassgn_pt1 ne 'unk'); 
+       			if hier(i) in ('010','011','110','111' '012','112' '013','113') then do; *Only want to grab those with procedure codes;
+         			Outassgn_pt1= vname(hier(i));
+       			end;
+     		end;
+
+			else if dx_n >=1 then do i=1 to 10 until (outassgn_pt1 ne 'unk');
+				*CDL: MODIFIED -- None should have procedure codes at this point, so a good check that working right;
+	       		*if hier(i) in ('100','110','101','111', '102','112', '103','113') then do;  *CDL: ADDED comma between 111 and 102;
+				if hier(i) in ('100','101','102','103') then do;
+	         		outassgn_pt1= vname(hier(i));
+	       		end;
+	     	end;
+
+	     	else if rx_n>=1 then do; *no need to loop since only uab has rx only;
+	         	outassgn_pt1= 'UAB';
+	     	end;
+	  	end;
+
+	 end;
+
+  	 *now step6 - 3b.6 - discordant abortion clean up;
+ 	 If outassgn_pt1 in ('SAB' 'IAB' 'UAB' ) then do;
+
+	 	*CDL: MODIFIED -- This was a bit cleaner to read and same results.;
+		if SAB ne ('000') and IAB ne ('000') then outassgn_pt2='UAB'; *Deal w discordant abortion outcomes first;
+				else if SAB ne ('000') then outassign_pt2='SAB';
+				else if IAB ne ('000') then outassign_pt2 = 'IAB';
+		*NOTE: Some may have meds for IAB or SAB because they had concordant codes on an encounter whihc were 
+				rolled up to the group level;
+
+/*		if sab ne '000' and iab ne '000' then outassgn_pt2='UAB'; */
+
+/*  		else do ;*3b.6.2 - go back to table 1 for abortion;*/
+/*         	if sab NE ('000') and uab NE ('000') then outassgn_pt2='SAB'; */
+/*    			else if iab NE ('000') and uab NE ('000') then outassgn_pt2='IAB';*/
+/*    			else if sab='000' and iab='000' and uab NE ('000') then outassgn_pt2='UAB';*/
+/*  		end;*/

```

Comment:
Changed the commented out version to the uncommented version because was easier to read.